xpallets/assets-registrar/src/lib.rs

```diff
@@ -271,36 +165,44 @@ decl_module! {
         /// Deregister an asset with given `id`.
         ///
         /// This asset will be marked as invalid.
+        ///
+        /// This is a root-only operation.
         #[weight = T::WeightInfo::deregister()]
         pub fn deregister(origin, #[compact] id: AssetId) -> DispatchResult {
             ensure_root(origin)?;
-            ensure!(Self::is_valid_asset(&id), Error::<T>::InvalidAsset);
+
+            ensure!(Self::asset_is_valid(&id), Error::<T>::AssetIsInvalid);
 
             AssetOnline::remove(id);
             T::RegistrarHandler::on_deregister(&id)?;
 
-            Self::deposit_event(RawEvent::Deregister(id));
+            Self::deposit_event(Event::Deregister(id));
 
             Ok(())
         }
 
         /// Recover a deregister asset to the valid state.
         ///
         /// `RegistrarHandler::on_register()` will be triggered again during the recover process.
+        ///
+        /// This is a root-only operation.
         #[weight = T::WeightInfo::recover()]
         pub fn recover(origin, #[compact] id: AssetId, has_mining_rights: bool) -> DispatchResult {
             ensure_root(origin)?;
+
             ensure!(Self::asset_exists(&id), Error::<T>::AssetDoesNotExist);
-            ensure!(!Self::is_valid_asset(&id), Error::<T>::AssetAlreadyValid);
+            ensure!(!Self::asset_is_valid(&id), Error::<T>::AssetAlreadyValid);
```

Comment:
Good reason for this change?

Reply:
just want to keep consistent with `ensure_asset_is_valid`

Reply:
That's not quite sound.

Reply:
I think this reason is enough.
If I rename the function `asset_exists` to `exists_asset` but keep the name of `ensure_asset_exists`. Isn't that weird?

Reply:
Just keep `is_valid_asset()`. From the point of naming an API, `is_valid_asset(asset_id)` is more common than `asset_is_valid(asset_id)` which is a bit of tedious, see https://doc.rust-lang.org/std/path/fn.is_separator.html .